File: solutions/scala/space-age/1/src/main/scala/SpaceAge.scala
// Dedicated to Shree DR.MDD

import scala.language.dynamics

object SpaceAge extends Dynamic {
  val orbitalMap: Map[String, Double] = Map(
    "onEarth"   -> 1.0,
    "onMercury" -> 0.2408467,
    "onVenus"   -> 0.61519726,
    "onMars"    -> 1.8808158,
    "onJupiter" -> 11.862615,
    "onSaturn"  -> 29.447498,
    "onUranus"  -> 84.016846,
    "onNeptune" -> 164.79132
  )

  def applyDynamic(planetIdentifier: String)(elapsedSeconds: Double): Double = {
    if (!(orbitalMap contains planetIdentifier)) throw new NoSuchMethodException
    "%.2f".format(elapsedSeconds / 31557600 / orbitalMap(planetIdentifier)).toDouble
  }

  def selectDynamic(unusedKey: String): Any =
    throw new NoSuchFieldException

  def updateDynamic(unusedKey: String)(value: Any): Any =
    throw new NoSuchFieldException
}

File: solutions/julia/space-age/1/space-age.jl
function onEarth(seconds)
    seconds / 31557600
end

function onMercury(seconds)
    onEarth(seconds) / 0.2408467 
end

function onVenus(seconds)
    onEarth(seconds) / 0.61519726 
end

function onMars(seconds)
    onEarth(seconds) / 1.8808158 
end

function onJupiter(seconds)
    onEarth(seconds) / 11.862615
end

function onSaturn(seconds)
    onEarth(seconds) / 29.447498 
end

function onUranus(seconds)
    onEarth(seconds) / 84.016846
end

function onNeptune(seconds)
    onEarth(seconds) / 164.79132
end